### PR TITLE
added email functionality.

### PR DESCRIPTION
**Pull request by kiransridhar**
created unit test for adding reviews and receiving email. edited acceptance test for adding review to include receiving emails

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -31,8 +31,11 @@ def create
     @review.site = @site
     @review.user = current_user
     if @review.save
+      @review.site.reviewed
```
Comment (dpickett) [inline app/controllers/reviews_controller.rb]: consider an instance method on review that "wraps this behavior" and mimics the return values of `save`

```
if @review.issue

# in app/models/review
def review
  if @review.save
    #send email
  else
    return false
  end
end
```

Then, write correlating unit tests to ensure this behaves as expected

File: app/mailers/review_added.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class ReviewAdded < ApplicationMailer
```
Comment (dpickett) [inline app/mailers/review_added.rb]: consider a more expressive mailer name. NewReviewNotification? ReviewNotification?

File: app/models/site.rb [hunk 3]
```diff
@@ -10,4 +10,16 @@ class Site < ActiveRecord::Base
   validates :phone, presence: true
   validates :description, presence: true
   validates :user_id, presence: true
+
+  def review(user)
```
Comment (dpickett) [inline app/models/site.rb]: is this being utilized? If not it should be removed.

File: app/models/site.rb [hunk 4]
```diff
@@ -10,4 +10,16 @@ class Site < ActiveRecord::Base
...
+    review.create!(comment: "Great coffee!", rating: 4, user_id: user.id)
+  end
+
+  def reviewed
+    # binding.pry
```
Comment (dpickett) [inline app/models/site.rb]: avoid commiting comment out binding.pry's - just delete them.

File: spec/models/user_adds_review_spec.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+require "rails_helper"
+RSpec.describe ReviewAdded, :type => :model do
+
+  describe "valid review" do
```
Comment (dpickett) [inline spec/models/user_adds_review_spec.rb]: this seems to be a bit confused between being a unit test and an acceptance test. You want your unit tests to exercise individual instance methods of your model

File: app/mailers/review_added.rb (hunk 2 above)
Comment (kiransridhar) [inline app/mailers/review_added.rb]: Will keep that in mind for the future. For now, to avoid confusion it was agreed to keep it the same

**Pull request merged**